### PR TITLE
Fixing bug with minimumOrigins=0

### LINKED ISSUE #1337: Attempts to create a pool with 0 origins will always fail
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the library and it is still present.

### cloudflare-go version

v0.72.0

### Go environment

GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/kelvin/Library/Caches/go-build"
GOENV="/Users/kelvin/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/kelvin/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/kelvin/go"
GOPRIVATE=""
GOROOT="/opt/homebrew/Cellar/go/1.20.5/libexec"
GOSUMDB="off"
GOTMPDIR=""
GOTOOLDIR="/opt/homebrew/Cellar/go/1.20.5/libexec/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.20.5"
GCCGO="gccgo"
AR="ar"
CC="cc"
CXX="c++"
CGO_ENABLED="1"
GOWORK=""
CGO_CFLAGS="-O2 -g"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/z1/4lpjdpw96m12m3xgtvvl3m1r0000gn/T/go-build158583462=/tmp/go-build -gno-record-gcc-switches -fno-common"

### Expected output

Pool created successfully

### Actual output

error: the minimum_origins must be in range [1, 0]: validation failed (1002)

### Code demonstrating the issue
```
package main

import (
	"context"
	"fmt"
	"github.com/cloudflare/cloudflare-go"
)

func main() {
	api, err := cloudflare.New("<redacted>", "<redacted>")
	if err != nil {
		panic(err)
	}
	accountId := cloudflare.AccountIdentifier("<redacted>")
	testPool := cloudflare.LoadBalancerPool{
		Description:    "Primary data center - Provider XYZ",
		Name:           "primary-dc-1",
		Enabled:        true,
		MinimumOrigins: 0,
		Origins:        []cloudflare.LoadBalancerOrigin{},
	}
	createPoolParams := cloudflare.CreateLoadBalancerPoolParams{LoadBalancerPool: testPool}
	pool, err := api.CreateLoadBalancerPool(context.Background(), accountId, createPoolParams)
	if err != nil {
		panic(err)
	}
	fmt.Printf("pool: %+v\n", pool)
}
```

### Steps to reproduce

Create a pool with no origins and MinimumOrigins set to 0

### References

_No response_

### PR DESCRIPTION
**Pull request by indeed-kelvin**
Provide a general summary of your changes in the title above. You should
remove this overview, any sections and any section descriptions you
don't need below before submitting. There isn't a strict requirement to
use this template if you can structure your description and still cover
these points.

## Description

Closes #1337

## Has your change been tested?

Confirmed pool gets created properly with the code snippet from issue #1337

## Types of changes

What sort of change does your code introduce/modify?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

Folks will have to use a int pointer for minimumOrigins instead of int for LoadBalancerPool.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

**Comment by jacobbednarz** [comment]
@tc80 are you able to confirm if this is something we intentionally don't support? i'm not sure why someone would want an empty pool but i may be missing something.

**Comment by tc80** [comment]
@jacobbednarz thanks for tagging me!

[From the API docs](https://developers.cloudflare.com/api/operations/load-balancer-pools-create-pool), `minimum_origins` is:
```
The minimum number of origins that must be healthy for this pool to serve traffic. If the number of healthy origins falls below this number, the pool will be marked unhealthy and will failover to the next available pool.
```
So a bit different from an "empty pool". I suppose if `minimum_origins=0`, then a pool will never be marked unhealthy and will always be able to serve traffic. I'll ask the team, but if this is allowed via the API then it should also be allowed via terraform.

File: load_balancing_test.go
Comment (tc80) [inline load_balancing_test.go:1]: Perhaps we should add an integration test that ensures we can create a pool when `MinimumOrigins=0`

**Comment by indeed-kelvin** [comment]
>  i'm not sure why someone would want an empty pool but i may be missing something.

To add some context, we are building a kubernetes operator to manage cloudflare objects and we have separate CRDs for pools and origins.  We want our users to be able to create pool objects in a cluster and then origin objects from any cluster.  Thus we have a need to be able to create pools with no origins so users can attach origins to them later.  We tested this behavior and found we can do so only if minimum_origins is set to 0.  If it is set to anything higher, we get the error: "the minimum_origins must be in range [1, 0]: validation failed" if we don't define any origins.  Thus not being able to set minimum_origins to 0 effectively prevents us from being able to create empty pools.

```
kelvin@IT-USA-VY1005 % curl --request POST -H "X-Auth-Key: $CF_API_KEY" \
  -H "X-Auth-Email: $CF_API_EMAIL" \
  -H "Content-Type: application/json" \
  "https://api.cloudflare.com/client/v4/accounts/<redacted>/load_balancers/pools" \
  --data '{
    "name": "pool-sample",
    "description": "pool-sample",
    "enabled": true,
    "latitude": 15,
    "longitude": 10,
    "minimum_origins": 1
  }'
{
  "result": null,
  "success": false,
  "errors": [
    {
      "code": 1002,
      "message": "the minimum_origins must be in range [1, 0]: validation failed"
    }
  ],
  "messages": []
}

kelvin@IT-USA-VY1005 % curl --request POST -H "X-Auth-Key: $CF_API_KEY" \
  -H "X-Auth-Email: $CF_API_EMAIL" \
  -H "Content-Type: application/json" \
  "https://api.cloudflare.com/client/v4/accounts/<redacted>/load_balancers/pools" \
  --data '{
    "name": "pool-sample",
    "description": "pool-sample",
    "enabled": true,
    "latitude": 15,
    "longitude": 10,
    "minimum_origins": 0
  }'
{
  "result": {
    "description": "pool-sample",
    "created_on": "2023-07-21T19:53:14.944732Z",
    "modified_on": "2023-07-21T19:53:14.944732Z",
    "id": "2cfcd8f564eaf40b6bd09195c122e21e",
    "enabled": true,
    "minimum_origins": 0,
    "name": "pool-sample",
    "notification_email": "",
    "check_regions": null,
    "latitude": 15,
    "longitude": 10,
    "origins": []
  },
  "success": true,
  "errors": [],
  "messages": []
}
```

File: load_balancing_test.go
Comment (indeed-kelvin) [reply load_balancing_test.go:1]: Done

**Comment by tc80** [comment]
talked with the team and this should be okay to merge 👍 
cc @gofish

**Review by tc80 — APPROVED** [review]

**Pull request merged**